src/app/server/RendezvousServer.cpp

```diff
@@ -59,11 +60,13 @@ void RendezvousServer::OnPlatformEvent(const DeviceLayer::ChipDeviceEvent * even
                          ChipError::FormatError(event->CommissioningComplete.status));
         }
         // TODO: Commissioning complete means we can finalize the admin in our storage
-    }
-    else if (event->Type == DeviceLayer::DeviceEventType::kOperationalNetworkEnabled)
-    {
-        app::Mdns::AdvertiseOperational();
-        ChipLogError(Discovery, "Operational advertising enabled");
+        break;
+    case DeviceLayer::DeviceEventType::kOperationalNetworkEnabled:
+    case DeviceLayer::DeviceEventType::kServiceProvisioningChange:
+        app::Mdns::StartServer();
```

Comment:
This change seems likely to break operational advertisement in various cases, due to https://github.com/project-chip/connectedhomeip/issues/7911

Reply:
Meh. Sounds like I need to add more ifdefs.

Reply:
Ok, reverted that change, but also changed conditions in `server/Mdns.cpp` as we discussed on slack.